tuplex/core/src/physical/TransformTask.cc

```diff
@@ -19,10 +19,17 @@
 
 namespace tuplex {
     // atomic var to count output rows!
-    static std::atomic_int64_t g_totalOutputRows;
+    static std::atomic_int64_t g_totalTopOutputRows;
+    static std::atomic_int64_t g_totalBottomOutputRows;
+
+    // mapping from order number -> row count if the task is finished
+    static std::mutex g_rowsDoneMutex;
+    static std::map<size_t, size_t> g_rowsDone;
 
     void TransformTask::resetOutputLimitCounter() {
```

Comment:
basically use htis here: => rename to resetLimits()

and give it the maximum order number for initlaizaton